controllers/osccluster_bastion_controller.go

```diff
@@ -0,0 +1,375 @@
+	if checkOscAssociate {
+		return nil
+	} else {
+		return fmt.Errorf("%s publicIp does not exist in bastion", bastionPublicIpName)
+	}
+}
+
+// checkBastionFormatParameters check Bastion parameters format.
+func checkBastionFormatParameters(clusterScope *scope.ClusterScope) (string, error) {
+	clusterScope.Info("Check Bastion parameters")
+	bastionSpec := clusterScope.GetBastion()
+	bastionSpec.SetDefaultValue()
+	bastionName := bastionSpec.Name + "-" + clusterScope.GetUID()
+	bastionTagName, err := tag.ValidateTagNameValue(bastionName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	imageName := bastionSpec.ImageName
+	if imageName != "" {
+		_, err := infrastructurev1beta1.ValidateImageName(imageName)
+		if err != nil {
+			return bastionTagName, err
+		}
+	} else {
+		_, err := infrastructurev1beta1.ValidateImageId(bastionSpec.ImageId)
+		if err != nil {
+			return bastionTagName, err
+		}
+	}
+
+	bastionKeypairName := bastionSpec.KeypairName
+	_, err = infrastructurev1beta1.ValidateKeypairName(bastionKeypairName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	vmType := bastionSpec.VmType
+	_, err = infrastructurev1beta1.ValidateVmType(vmType)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionDeviceName := bastionSpec.DeviceName
+	_, err = infrastructurev1beta1.ValidateDeviceName(bastionDeviceName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionSubregionName := bastionSpec.SubregionName
+	_, err = infrastructurev1beta1.ValidateSubregionName(bastionSubregionName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionSubnetName := bastionSpec.SubnetName
+	clusterScope.Info("Get bastionSubnetName", "bastionSubnetName", bastionSubnetName)
+
+	ipSubnetRange := clusterScope.GetIpSubnetRange(bastionSubnetName)
+	bastionPrivateIps := clusterScope.GetBastionPrivateIps()
+	var subnetsSpec []*infrastructurev1beta1.OscSubnet
+	networkSpec := clusterScope.GetNetwork()
+	networkSpec.SetSubnetDefaultValue()
+	subnetsSpec = networkSpec.Subnets
+	for _, subnetSpec := range subnetsSpec {
+		subnetName := subnetSpec.Name
+		ipSubnetRange := subnetSpec.IpSubnetRange
+		clusterScope.Info("Get IpSubnetRange", "ipSubnetRange", ipSubnetRange)
+		clusterScope.Info("Get subnetName", "subnetName", subnetName)
+	}
+	for _, bastionPrivateIp := range *bastionPrivateIps {
+		privateIp := bastionPrivateIp.PrivateIp
+		clusterScope.Info("### Get Valid IP ###", "privateIp", privateIp)
+		clusterScope.Info("### Get Valid subnet ###", "ipSubnetRange", ipSubnetRange)
+		_, err := compute.ValidateIpAddrInCidr(privateIp, ipSubnetRange)
+		if err != nil {
+			return bastionTagName, err
+		}
+	}
+
+	if bastionSpec.RootDisk.RootDiskIops != 0 {
+		rootDiskIops := bastionSpec.RootDisk.RootDiskIops
+		clusterScope.Info("Check rootDiskIops", "rootDiskIops", rootDiskIops)
+		_, err := infrastructurev1beta1.ValidateIops(rootDiskIops)
+		if err != nil {
+			return bastionTagName, err
+		}
```

Comment:
I recently discovered that the ratio iops/size should not execeed 300, mayeb check that in the future

Reply:
See https://docs.outscale.com/en/userguide/About-Volumes.html#_volume_types_and_iops

Reply:
Ok good idea:
https://github.com/outscale/cluster-api-provider-outscale/issues/216